src/api/app_agent/websocket.ts

```diff
@@ -0,0 +1,94 @@
+/**
+ * Defines AppAgentWebsocket, an easy-to-use websocket implementation of the
+ * Conductor API for apps, restricted to a single app provided on initialization
+ *
+ *    const appWs = AppWebsocket.connect(
+ *      'ws://localhost:9000',
+ *      signal => console.log('got a signal:', signal)
+ *    )
+ *
+ *    const client = new AppAgentWebsocket(appWs, 'my_installed_app_id')
+ * 
+ *    client.callZome({
+ *      role_id: 'my_role_id' // role_id is unique per app, so you can unambiguously identify your dna with role_id in this client,
+ *      zome_name: 'zome',
+ *      fn_name: 'fn',
+ *      payload: { value: 'v' }
+ *    })
+ *      .then(result => {
+ *        console.log('callZome returned with:', result)
+ *      })
+ *      .catch(err => {
+ *        console.error('callZome errored with:', err)
+ *      })
+ */
+
+import { EventEmitter } from 'events'
+import ldsh from 'lodash'
+const { omit } = ldsh
+
+import { InstalledAppId } from '../../types.js';
+import { AppInfoResponse, AppWebsocket, ArchiveCloneCellResponse, CallZomeRequest, CallZomeResponse, CreateCloneCellRequest, CreateCloneCellResponse, InstalledAppInfo } from "../index.js";
+import {
+  AppAgentCallZomeRequest,
+  AppAgentClient,
+  AppArchiveCloneCellRequest,
+  AppCreateCloneCellRequest,
+} from "./types.js";
+
+export class AppAgentWebsocket extends EventEmitter implements AppAgentClient  {
+  appWebsocket: AppWebsocket;
+  installedAppId: InstalledAppId;
+
+  constructor(
+    appWebsocket: AppWebsocket,
+    installedAppId: InstalledAppId
+  ) {
+    super()
+    this.appWebsocket = appWebsocket;
+    this.installedAppId = installedAppId
+
+    this.appWebsocket.on('signal', signal => this.emit('signal', signal))
```

Comment:
Currently this forwards ALL signals, it doesn't filter based on `installed_app_id`. I didn't add that filter, as it would require an additional call to `appInfo` for every signal, but it's possible it's worth it.

Curious your thoughts on this.